control-plane/go.mod

```diff
@@ -1,5 +1,8 @@
 module github.com/hashicorp/consul-k8s/control-plane
 
+// TODO: Remove this when the next version of the submodule is released.
+replace github.com/hashicorp/consul/proto-public => github.com/hashicorp/consul/proto-public v0.1.2-0.20231109213314-40c57f10a0fb
+
```

Comment:
I think we've typically targeted SHA versions as direct dependencies, rather than via `replace` directives, when an unreleased version of a sub is needed (there's a few recent examples in git history). ❓ Is there a reason for using `replace` in this case?

FWIW, I think @thisisnotashwin is planning to do something similar in https://github.com/hashicorp/consul-k8s/pull/3188/files#diff-b85c459442bcfeb4bb66362e683f1648d6d097e9478a95a43c40bffc82769b4aR14

Reply:
That's fine, I thought a replace model was easier to remove when we get to the next submodule release.

Reply:
`replace` has been used most recently -- I undid it as part of the 1.3.0 release. The reason was lost from the comment that previously explained how `/api` will overwrite `proto-public` if you just do a simple pin

https://github.com/hashicorp/consul-k8s/commit/0f81470bde7c00d6da419068ef23a15f60e436cb#diff-b85c459442bcfeb4bb66362e683f1648d6d097e9478a95a43c40bffc82769b4a

Reply:
Oh yeah, if you just pin right now and then do a `tidy` it will clobber the pinned version and you will get this: 

<img width="576" alt="image" src="https://github.com/hashicorp/consul-k8s/assets/29112081/8fa811cb-f2eb-498e-9693-b658b48d9a97">

Reply:
Ah, I forgot about that comment. Thanks y'all! Appreciate the restoration for clarity (clearly I need it... 😆)